### PR TITLE
Deprecate instanceId Config For Broker/Minion Specific Configs

### LINKED ISSUE #9290: Pinot Broker Uses a Generic "instanceId" Property Which is Shared With Minion
At present BaseBrokerStarter determines the Broker Instance ID using a very generic "instanceId" property which is also shared with Pinot minion. In the case of a Quickstart Cluster, if we want to start the Broker with a particular instanceId the start-up fails since the same property is used by the Minion as well.

I think we can change this logic to the following (use pinot.broker.id over instanceId):

```
    _instanceId = _brokerConf.getProperty(Broker.CONFIG_OF_BROKER_ID);
    if (_instanceId == null && _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY) != null) {
        _instanceId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY);
    }
    if (_instanceId != null) {
      // NOTE: Force all instances to have the same prefix in order to derive the instance type based on the instance id
      Preconditions.checkState(InstanceTypeUtils.isBroker(_instanceId), "Instance id must have prefix '%s', got '%s'",
          Helix.PREFIX_OF_BROKER_INSTANCE, _instanceId);
    } else {
      _instanceId = Helix.PREFIX_OF_BROKER_INSTANCE + _hostname + "_" + _port;
    }

    _brokerConf.setProperty(Broker.CONFIG_OF_BROKER_ID, _instanceId);
```

https://github.com/apache/pinot/blob/53c117f50194754aec3ecbcae0e065d0901c0712/pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java#L136:L145

### PR DESCRIPTION
**Pull request by ankitsultana**
fixes #9290

I felt it might be best to mark instanceId as deprecated and create a new config for Minion ID. Lmk your thoughts.

cc: @Jackie-Jiang

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java [hunk 1]
```diff
@@ -133,14 +133,15 @@ public void init(PinotConfiguration brokerConf)
...
-    _instanceId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY);
-    if (_instanceId != null) {
-      // NOTE: Force all instances to have the same prefix in order to derive the instance type based on the instance id
-      Preconditions.checkState(InstanceTypeUtils.isBroker(_instanceId), "Instance id must have prefix '%s', got '%s'",
-          Helix.PREFIX_OF_BROKER_INSTANCE, _instanceId);
-    } else {
+    _instanceId = _brokerConf.getProperty(Broker.CONFIG_OF_BROKER_ID);
+    if (_instanceId == null && _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY) != null) {
+      _instanceId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY);
+    } else if (_instanceId == null) {
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java:139]:
```suggestion
    _instanceId = _brokerConf.getProperty(Broker.CONFIG_OF_BROKER_ID);
    if (_instanceId == null) {
      _instanceId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY);
    }
    if (_instanceId == null) {
```

File: pinot-minion/src/main/java/org/apache/pinot/minion/MinionConf.java [hunk 2]
```diff
@@ -58,6 +59,9 @@ public int getPort() {
   }
 
   public String getInstanceId() {
+    if (StringUtils.isNotBlank(getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID))) {
+      return getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID);
+    }
```
Comment (Jackie-Jiang) [inline pinot-minion/src/main/java/org/apache/pinot/minion/MinionConf.java:64]: (minor) maybe keeping the behavior consistent, and do null check for all of them
```suggestion
    String instanceId = getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID);
    return instanceId != null ? instanceId : getProperty(CommonConstants.Helix.Instance.INSTANCE_ID_KEY);
```

**Review by Jackie-Jiang — COMMENTED** [review]
We can deprecate `instanceId`

Currently controller and server is using `pinot.<type>.instance.id` as the key, I'd suggest making broker and minion following the same pattern to be consistent

**Comment by ankitsultana** [comment]
> Currently controller and server is using pinot.<type>.instance.id as the key, I'd suggest making broker and minion following the same pattern to be consistent

@Jackie-Jiang : We might need to deprecate pinot.broker.id as well in that case. Is that okay?

**Comment by Jackie-Jiang** [comment]
> > Currently controller and server is using pinot..instance.id as the key, I'd suggest making broker and minion following the same pattern to be consistent
> 
> @Jackie-Jiang : We might need to deprecate pinot.broker.id as well in that case. Is that okay?

@ankitsultana Per the current logic, I think we can directly modify the `CONFIG_OF_BROKER_ID` to `pinot.broker.instance.id` without worrying about backward compatibility because the field is always set within the `BaseBrokerStarter` and then read again

**Comment by ankitsultana** [comment]
The field is read in `BrokerAdminApiApplication` and `BaseBrokerRequestHandler`.

But you are right, I took a look and both of these classes get the value from the one set by `BaseBrokerStarter`. I'll update the PR.

**Comment by ankitsultana** [comment]
@Jackie-Jiang : Updated. One of the integration tests is failing with a "ServerSegmentMissing" error. Not sure if it is related.

**Review by Jackie-Jiang — APPROVED** [review]
LGTM. Let me re-run the failed test

**Pull request merged**